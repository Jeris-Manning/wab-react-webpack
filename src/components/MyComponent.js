import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Header } from 'semantic-ui-react';
import styled from 'styled-components';

import { withWidgetContext } from './Context';
import MyChildComponent from './MyChildComponent';

import loadLayers from '../utils/loadLayers';

const Padding = styled.div`
  padding: 0.25rem;
`;

class MyComponent extends Component {
  state = {
    msg: 'WAB',
    layers: null,
    loading: true,
    error: false,
  };

  componentDidMount() {
    this.loadLayers();

    this.updateMsg();
  }

  updateMsg = () => {
    setInterval(() => {
      const { msg } = this.state;
      const newMsg = msg === 'HELLO' ? 'WAB' : 'HELLO';
      this.setState({ msg: newMsg });
    }, 3000);
  };

  loadLayers = async () => {
    // Load Layers
    const { wab, esriJS } = this.props;
    console.log(this.props);
    try {
      const layers = await loadLayers(
        esriJS,
        wab.map,
        wab.config.layerCollection,
      );
      this.setState({ layers, loading: false });
    } catch (error) {
      this.setState({ error, loading: false });
    }
  };

  render() {
    if (this.state.loading) {
      return <div>Loading...</div>;
    }
    return (
      <Padding>
        <Header>{this.props.title}</Header>
        <MyChildComponent msg={this.state.msg} />
      </Padding>
    );
  }
}

MyComponent.propTypes = {
  title: PropTypes.string,
  wab: PropTypes.object,
  esriJS: PropTypes.object,
};

MyComponent.defaultProps = {
  title: 'Hello',
  wab: {},
  esriJS: {},
};

export default withWidgetContext()(MyComponent);
