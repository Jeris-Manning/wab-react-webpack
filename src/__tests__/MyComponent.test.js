import React from 'react';
import 'react-testing-library/cleanup-after-each';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import MyComponent from '../components/MyComponent';

afterEach(cleanup);

// https://testing-library.com/docs/example-react-context
describe('<MyComponent />', () => {
  it('renders main component without error', () => {
    const { container, getByText } = render(<MyComponent />);
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('WAB')).toBeInTheDocument();
  });
});
