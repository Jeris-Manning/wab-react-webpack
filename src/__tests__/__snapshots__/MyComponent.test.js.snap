// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> renders main component without error 1`] = `
<div
  class="sc-bdVaJa ldkUXY"
>
  <div
    class="ui header"
  >
    Hello
  </div>
  <div
    class="ui segment"
  >
    <h3>
      WAB
    </h3>
    <h5>
      Widget Open
    </h5>
  </div>
  <div
    class="sc-bwzfXH gXWnte"
  >
    Version: 
  </div>
</div>
`;
